sandbox/data_science/app_files/text_to_skills.py

```diff
@@ -0,0 +1,235 @@
+
+    # remove any remaining stopwords
+    trigram_review = [term for term in trigram_doc
+                      if not term in stopwords]
+
+    # create a bag-of-words representation
+    doc_bow = trigram_dictionary_POS.doc2bow(trigram_doc)
+
+    # create an LDA representation
+    document_lda = lda[doc_bow]
+    return trigram_review, document_lda
+
+def lda_top_topics(document_lda, topic_names, min_topic_freq=0.05):
+    '''
+    Print a sorted list of the top topics for a given LDA representation
+    '''
+    # sort with the most highly related topics first
+    sorted_doc_lda = sorted(document_lda, key=lambda review_lda: -review_lda[1])
+
+    for topic_number, freq in sorted_doc_lda:
+        if freq < min_topic_freq:
+            break
+
+        # print the most highly related topic names and frequencies
+        print('*'*56)
+        print('{:50} {:.3f}'.format(topic_names[topic_number+1],
+                                round(freq, 3)))
+        print('*'*56)
+        for term, term_freq in lda.show_topic(topic_number, topn=10):
+            print(u'{:20} {:.3f}'.format(term, round(term_freq, 3)))
+        print('\n\n')
+
+def top_match_items(document_lda, topic_names, num_terms=100):
+    '''
+    Print a sorted list of the top topics for a given LDA representation
+    '''
+    # sort with the most highly related topics first
+    sorted_doc_lda = sorted(document_lda, key=lambda review_lda: -review_lda[1])
+
+    topic_number, freq = sorted_doc_lda[0][0], sorted_doc_lda[0][1]
+    print('*'*56)
+    print('{:50} {:.3f}'.format(topic_names[topic_number+1],
+                            round(freq, 3)))
+    print('*'*56)
+    for term, term_freq in lda.show_topic(topic_number, topn=num_terms):
+        print(u'{:20} {:.3f}'.format(term, round(term_freq, 3)))
+
+
+def top_match_list(document_lda, topic_names, num_terms=100):
+    # Take the above results and just save to a list of the top 500 terms in the topic
+    sorted_doc_lda = sorted(document_lda, key=lambda review_lda: -review_lda[1])
+    topic_number, freq = sorted_doc_lda[0][0], sorted_doc_lda[0][1]
+    print('Highest probability topic:', topic_names[topic_number+1],
+                                         '\t', round(freq, 3))
+    top_topic_skills = []
+    for term, term_freq in lda.show_topic(topic_number, topn=num_terms):
+        top_topic_skills.append(term)
+    return top_topic_skills
+
+def common_skills(top_topic_skills, user_skills):
+    return [item for item in top_topic_skills if item in user_skills]
+
+def non_common_skills(top_topic_skills, user_skills):
+    return [item for item in top_topic_skills if item not in user_skills]
+
+def output_all_skills(text_document, num_skills):
+    user_skills, my_lda = vectorize_input(text_document, bigram_model, trigram_model, trigram_dictionary)
+    skills_list = top_match_list(my_lda, topic_names, num_terms=500)
+    print("Top 20 skills user has in common with topic:")
```

Comment:
this is still hard-coded to 20 even thought you exposed `num_skills` as a parameter.

change to:

```
msg = "Top {} skills user has in common with topic:"
print(msg.format(num_skills))
```

Reply:
:+1: this is just an oversight